### PR TITLE
chore: [M3-8262] - Cleanup eventMessages flag and legacy code

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This PR is a follow up to EventMessages v2 implementation. Now that the flag has been `on` in production and no major (or minor for that regard) bugs has been reported, it's time to clean up the legacy code and ordganize the code around the features EventMessages v2 introduced and implemented.

This refactor was facilited by targeting all instances of `TODO eventMessagesV2` left in the codebase to this purpose.

**This PR does not modify behavior or styling in any way** (read _shouldn't_), it only reorganizes the code around the cleanup.

## Changes  🔄
- Delete eventMessagesV2 flag
- Delete Events and NotificationCenter legacy code
- Update NotificationCenter structure/naming conventions
- Update tests (unit & e2e)

## Preview 📷
**Include a screenshot or screen recording of the change**

:bulb: Use `<video src="" />` tag when including recordings in table.

| NotificationCenter  | Events  |
| ------- | ------- |
| Screenshot 2024-08-29 at 13 12 10 | Screenshot 2024-08-29 at 13 11 30 |

## How to test 🧪
Verify that no regression happens as a result of this cleanup

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/NotificationCenter/Notifications/NotificationCenterNotifications.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                  [classes.inverted]: showAll,
+                })}
+              />
+            </StyledLinkButton>
+          </StyledLToggleContainer>
+        ) : null}
+      </>
+    );
+  }
+);
```
Comment (abailly-akamai) [inline packages/manager/src/features/NotificationCenter/Notifications/NotificationCenterNotifications.tsx:90]: This and underneath isn't new code, it's just been broken down - logic is the same

File: packages/manager/src/features/NotificationCenter/NotificationCenter.styles.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      left: 0,
+      position: 'absolute',
+      top: 0,
+      width: 4,
+    },
+    backgroundColor: theme.bg.offWhite,
+    borderBottom: `1px solid ${theme.bg.main}`,
+    position: 'relative',
+  },
+}));
```
Comment (abailly-akamai) [inline packages/manager/src/features/NotificationCenter/NotificationCenter.styles.ts:149]: Following our standards, broke styles and types into own files

no new code here

File: packages/manager/src/features/Events/constants.ts [hunk 3]
```diff
@@ -155,6 +24,19 @@ export const ACTIONS_TO_INCLUDE_AS_PROGRESS_EVENTS: Event['action'][] = [
...
+  'entity_transfer_cancel',
+  'entity_transfer_create',
+  'entity_transfer_fail',
+  'entity_transfer_stale',
+  'lassie_reboot',
+  'community_like',
+  'community_mention',
+  'community_question_reply',
+];
+
```
Comment (abailly-akamai) [inline packages/manager/src/features/Events/constants.ts:39]: this was moved from a file that was deleted (this block was added after the initial refactor)

File: packages/manager/src/features/Events/EventsLanding.tsx [hunk 4]
```diff
@@ -105,19 +94,12 @@ export const EventsLanding = (props: Props) => {
...
-                <TableCell data-qa-events-username-header sx={{ width: 150 }}>
-                  User
-                </TableCell>
-              </Hidden>
-            )}
+            <Hidden smDown>
+              <TableCell data-qa-events-username-header sx={{ width: 150 }}>
+                User
+              </TableCell>
+            </Hidden>
```
Comment (mjac0bs) [inline packages/manager/src/features/Events/EventsLanding.tsx:102]: Not really in scope of this PR, but: this column is a slightly different color than the others. Or, rather, it's the color that matches other table headers throughout the app, but the non-Hidden `StyledLabelTableCells` use `theme.textColors.tableHeader`, which is especially low contrast in dark mode.

lightmode

darkmode

Fun fact: StackScripts are like this too, but at least every column is the same. 😅 
stackscripts

File: packages/manager/src/features/NotificationCenter/utils.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+export const isEUModelContractNotification = (notification: Notification) => {
+  return (
+    notification.type === 'notice' && /eu-model/gi.test(notification.message)
+  );
+};
+
+// For communicative purposes in the UI, in some cases we want to adjust the severity of certain notifications compared to what the API returns.
+// If it is a maintenance notification of any sort, we display them as major instead of critical. Otherwise, we return the existing severity.
+export const adjustSeverity = ({
```
Comment (mjac0bs) [inline packages/manager/src/features/NotificationCenter/utils.ts:24]: Can we make this a JSDoc comment, while we're here?

**Review by mjac0bs — APPROVED** [review]
Congrats on the clean up, look at that diff. 🎉 (We should remember to archive the LD flag after the release goes out too.)

Reorganizations and renamings make sense! Long, but clear and consistent, component names. 👍🏼 

Didn't see any regressions after testing some events locally. CI had some trouble with a couple runners, but local runs of those tests looked good.

File: packages/manager/src/features/NotificationCenter/utils.ts (hunk 5 above)
Comment (abailly-akamai) [reply packages/manager/src/features/NotificationCenter/utils.ts]: yup done

File: packages/manager/src/features/Events/EventsLanding.tsx (hunk 4 above)
Comment (abailly-akamai) [reply packages/manager/src/features/Events/EventsLanding.tsx:102]: good 👀 ! fixed

**Comment by abailly-akamai** [comment]
> Long, but clear and consistent, component names. 

@mjac0bs yeah i struggled a bit with a decent convention but "Notification" in itself isn't conveying enough context. My take was that the names here makes it easier to find references for contributors in their IDE, and by compile time/minification the name length matters little.

**Review by bnussman-akamai — APPROVED** [review]

**Pull request merged**